### PR TITLE
Replace -s -d -n with --age argument

### PR DESCRIPTION
**Pull request by mkosiarc**
Fixes https://github.com/redhat-aqe/review-rot/issues/67, But instead of using --since and --until we decided to use --age.

`review-rot --age older 5d 10h `
outputs MRs/PRs which ale older than 5 days and 10 hours,
`review-rot --age newer 5d 10h `
outputs MRs/PRs which are newer than 5 days and 10 hours.

### CODE REVIEW

**Comment by pbortlov** [comment]
@danrodrig please review this

File: reviewrot/basereview.py [hunk 1]
```diff
@@ -10,83 +10,32 @@
...
-                """ use abs_diff to calculate absolute time difference
-                in minutes """
-                abs_min = abs_diff.total_seconds() / 60
-                if state_ == 'older' and abs_min < value:
-                    return False
-                elif state_ == 'newer' and abs_min >= value:
-                    return False
-            else:
-                raise ValueError("Invalid duration type: %s" % duration)
+        if age is not None:
```
Comment (amarza-rh) [inline reviewrot/basereview.py]: This is a bit hard to parse visually. I would suggest:

```python
# early exit
if age is None:
    return True

if age.state == 'newer' and created_at > age.date:
    return True

return False
```

I would also add a unit test for each of these branches.

Comment (mkosiarc) [reply reviewrot/basereview.py]: thanks, but this PR is not yet ready, I had to abandon it for some time, there are some things missing, also tests and docs.

**Comment by mkosiarc** [comment]
@pbortlov @amarza-rh please review

**Comment by pbortlov** [comment]
@danrodrig could you please do a review?

**Comment by amarza-rh** [comment]
I see some refactoring happening here together with the changes.

This makes this review harder than necessary. Can you split these changes into two or more commits? i.e. one with the changes needed and another for the refactoring involved. You can also leave out unnecessary refactoring for a subsequent PR.

Also specify testing done, with expected / actual output if you can.

Thanks!

File: README.md [hunk 2]
```diff
@@ -73,7 +68,18 @@ SSL:
...
                         verification
 
+
+```
+
+You can filter MRs/PRs based on their relative age
+```
+review-rot --age older 5d 10h
+```
+outputs MRs/PRs which ale older than 5 days and 10 hours,
```
Comment (danrodrig) [inline README.md]: typo! ...which are older...

File: README.md [hunk 3]
```diff
@@ -73,7 +68,18 @@ SSL:
...
+```
+review-rot --age newer 5d 10h
 ```
+outputs MRs/PRs which are newer than 5 days and 10 hours
```
Comment (danrodrig) [inline README.md]: Same here, submitted in the last 5 days and 10 hours

File: reviewrot/__init__.py [hunk 4]
```diff
@@ -1,20 +1,32 @@
...
 from reviewrot.gitlabstack import GitlabService
 from reviewrot.pagurestack import PagureService
+from reviewrot.basereview import Age
 
 import yaml
 
 log = logging.getLogger(__name__)
 
+# Valid values of choices for arguments
+FORMAT_CHOICES = ['oneline', 'indented', 'json']
```
Comment (danrodrig) [inline reviewrot/__init__.py]: I would do FORMAT_CHOICES and SORT_CHOICES or CHOICES = {'format': [...], 'sort': [...]}, but not both.

**Comment by danrodrig** [comment]
As @amarza-rh commented, would be better to split the PR.

File: README.md (hunk 2 above)
Comment (mkosiarc) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (mkosiarc) [reply README.md]: fixed

File: reviewrot/__init__.py (hunk 4 above)
Comment (mkosiarc) [reply reviewrot/__init__.py]: fixed

**Comment by mkosiarc** [comment]
@amarza-rh @danrodrig 
I removed some unecessary refactoring. The tests relevant for this PR are:

-  ParseAgeTest class, which tests parse static method from ParseAge class located in __init__py.
- BaseServiceCheckRequestStateTest class, which tests  check_request_state method from BaseService class located in basereview.py
- test_age_argument_in_command_line_valid, test_age_argument_in_config from CommandLineParserTest

File: reviewrot/basereview.py [hunk 5]
```diff
@@ -10,84 +10,30 @@
...
-                abs_min = abs_diff.total_seconds() / 60
-                if state_ == 'older' and abs_min < value:
-                    return False
-                elif state_ == 'newer' and abs_min >= value:
-                    return False
-            else:
-                raise ValueError("Invalid duration type: %s" % duration)
-        return True
+
+        if age is None:
```
Comment (pbortlov) [inline reviewrot/basereview.py]: YOu can add line 30-31 to line 33

File: README.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ### New: 
 - Phabricator support (check examples/sampleinput_phabricator.yaml )
 - Possibility of omitting WIP pull requests/merge requests in output with *--ignore-wip* argument
+- Replace *-s, -v, -d* arguments with one argument *--age*
```
Comment (pbortlov) [inline README.md]: The newest should be at the top of the list.

File: reviewrot/basereview.py (hunk 5 above)
Comment (danrodrig) [reply reviewrot/basereview.py]: I think both are fine. It's a personal preference I guess. Mine would be:

```python
if age is None:
    return True

if age.state == 'newer':
    return created_at > age.date
elif age.state == 'older':
    return created_at < age.date
else:
    raise ValueError(f"Wrong value for age state: {age.state}")
```
